File: in_java/Pattern_8.java
package in_java;

public class Pattern_8 extends Pattern {

    public Pattern_8(int row, int column, String symbol) {
        super(row, column, symbol);
    }

    public void print() {
        int symbol = Integer.parseInt(this.symbol);
        for (int i = 0; i < this.row; i++) {
            symbol = Integer.parseInt(this.symbol);
            for (int j = 0; j < this.column; j++) {
                if (j >= row - 1 - i && j <= row - 1 + i) {
                    System.out.print(symbol);
                    if (j < row - 1) {
                        symbol++;
                    } else {
                        symbol--;
                    }
                } else {
                    System.out.print(" ");
                }
            }
            System.out.println();
        }
    }
}

File: in_rust/src/pattern_8.rs
pub fn pattern_8() {
    let row = 5;
    let column = 9;
    let mut symbol;

    for i in 0..row {
        symbol = 1;
        for j in 0..column {
            if j >= row - 1 - i && j <= row - 1 + i {
                print!("{}", symbol);
                if j < row - 1 {
                    symbol += 1;
                } else {
                    symbol -= 1;
                }
            } else {
                print!(" ");
            }
        }
        println!();
    }
}
